src/main/java/org/folio/service/auth/ApiClient.java

```diff
@@ -0,0 +1,184 @@
+package org.folio.service.auth;
+
+import static org.folio.dataimport.util.RestUtil.OKAPI_TENANT_HEADER;
+import static org.folio.dataimport.util.RestUtil.OKAPI_TOKEN_HEADER;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vertx.core.json.JsonObject;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.ws.rs.core.MediaType;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dataimport.util.OkapiConnectionParams;
+
+public abstract class ApiClient {
```

Comment:
Did you copy this class from somewhere? or write it yourself? I ask cause i think there might be something in RMB/elsewhere that does the same thing.

Reply:
I got the idea from @TarasSpashchenko , who pointed me to [similar code in mod-oai-pmh](https://github.com/folio-org/mod-oai-pmh/blob/master/src/main/java/org/folio/oaipmh/helpers/client/InventoryClient.java).  When we were discussing, there were two other alternatives I found:
- The builtin RMB method would be to alter the other packages (users, perms, etc) to have them generate and expose additional code/packages, such as how mod-source-record-manager does it.  This would require relatively significant breaking changes in the other modules and was deemed far too much.
- The other common way I've seen in FOLIO is using something like Feign, which is part of the Spring Way.  However, retrofitting this existing RMB/Vert.x module with all the required dependencies and infrastructure for Feign would have been a major investment and also require some higher-level changes within DI.

We decided adding basic clients like this was sufficiently straighforward without major repercussions.

Reply:
sounds good.